assets/src/edit-story/app/story/actions/test/useAutoSave.js

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { fireEvent, render } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import APIContext from '../../../api/context';
+import ConfigContext from '../../../config/context';
+import useAutoSave from '../useAutoSave';
+import getStoryMarkup from '../../../../utils/getStoryMarkup';
+
+jest.mock('../../../../utils/getStoryMarkup', () => jest.fn());
+
+function setup(args) {
+  let result = {};
```

Comment:
@barklund Let me know if there's a better way to test this.

Reply:
We have `@testing-library/react-hooks` in the stack. Using that makes this a lot simpler. You can still create proper wrapper contexts, but don't need to manually create a dummy content to contain the hook - the library does that for you using [`renderHook`](https://github.com/testing-library/react-hooks-testing-library#usecountertestjs).

Reply:
I tried it initially with `renderHook` actually but didn't succeed (haven't used it before with needing data from all the other contexts). I guess I didn't use it properly, will try again. I might send you a diff to ask what's wrong.

Reply:
OK, found what I was missing before. Updated to use `renderHook` now.